daprdocs/content/en/developing-applications/integrations/cloud-providers/authenticating-azure.md

```diff
@@ -0,0 +1,399 @@
+
+{{% /codetab %}}
+
+{{% codetab %}}
+If you'd rather use a **PFX certificate**, run this command which will create a self-signed certificate:
+
+```sh
+az ad app credential reset \
+  --id $APP_ID \
+  --create-cert
+```
+
+> Note: self-signed certificates are recommended for development only. For production, you should use certificates signed by a CA and imported with the `--cert` flag.
+
+The output of the command above should look like:
+
+```json
+{
+  "appId": "c7dd251f-811f-4ba2-a905-acd4d3f8f08b",
+  "fileWithCertAndPrivateKey": "/Users/alessandro/tmpgtdgibk4.pem",
+  "name": "c7dd251f-811f-4ba2-a905-acd4d3f8f08b",
+  "password": null,
+  "tenant": "cd4b2887-304c-47e1-b4d5-65447fdd542b"
+}
+```
+
+Take note of the values above, which you'll need to use in your Dapr components' metadata:
+
+- `appId` is the value for `azureClientId`
+- `tenant` is the value for `azureTenantId`
+- The self-signed PFX certificate and private key are written in the file at the path specified in `fileWithCertAndPrivateKey`.  
+  Use the contents of that file as `azureCertificate` (or write it to a file on the server and use `azureCertificateFile`)
+
+{{% /codetab %}}
+
+{{< /tabs >}}
+
+Once you have created an Azure AD application, we need to create a Service Principal for that application, which will allow us to grant it access to Azure resources. Run:
+
+```sh
+SERVICE_PRINCIPAL_ID=$(az ad sp create \
+  --id $APP_ID \
+  | jq -r .objectId)
+echo "Service Principal ID: ${SERVICE_PRINCIPAL_ID}"
+```
+
+The output will be similar to:
+
+```text
+Service Principal ID: 1d0ccf05-5427-4b5e-8eb4-005ac5f9f163
+```
+
+Note that the value above is the ID of the **Service Principal** which is different from the ID of application in Azure AD (client ID)! The former is defined within an Azure tenant and is used to grant access to Azure resources to an application. The client ID instead is used by your application to authenticate. To sum things up:
+
+- You'll use the client ID in Dapr manifests to configure authentication with Azure services
+- You'll use the Service Principal ID to grant permissions to an application to access Azure resources
+
+Keep in mind that the Service Principal we just created does not have access to any Azure resource by default. Access will need to be granted to each resource as needed, as documented in the docs for the components.
+
+> Note: this step is different from the [official documentation](https://docs.microsoft.com/en-us/cli/azure/create-an-azure-service-principal-azure-cli) as the short-hand commands create a Service Principal that has broad read-write access to all Azure resources in your subscription. Not only this grants our application more access than you are likely going to desire, but this also applies only to the Azure management plane (Azure Resource Manager, or ARM), which is irrelevant for Dapr anyways (all Azure components are designed to interact with the data plane of various services, and not ARM).
```

Comment:
you can pass the scope you want during the creation of the SP using the --scopes parameter

Reply:
See my comment above. The SP shouldn't have any access at the subscription level, it only needs data plane access and that should be granted at the resource level (for the principle of the least privileges)